source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -105,6 +107,67 @@ EM_BOOL CIrrDeviceSDL::MouseLeaveCallback(int eventType, const EmscriptenMouseEv
 }
 #endif
 
+
+bool CIrrDeviceSDL::keyIsKnownSpecial(EKEY_CODE key) const
+{
+	switch ( key )
+	{
+		// keys which are known to have safe special character interpretation
+		// could need changes over time (removals and additions!)
+		case KEY_RETURN:
+		case KEY_PAUSE:
+		case KEY_ESCAPE:
+		case KEY_PRIOR:
+		case KEY_NEXT:
+		case KEY_END:
+		case KEY_HOME:
+		case KEY_LEFT:
+		case KEY_UP:
+		case KEY_RIGHT:
+		case KEY_DOWN:
+		case KEY_PRINT:
+		case KEY_SNAPSHOT:
+		case KEY_INSERT:
+		case KEY_BACK:
+		case KEY_DELETE:
+		case KEY_HELP:
+		case KEY_APPS:
+		case KEY_SLEEP:
+		case KEY_F1:
+		case KEY_F2:
+		case KEY_F3:
+		case KEY_F4:
+		case KEY_F5:
+		case KEY_F6:
+		case KEY_F7:
+		case KEY_F8:
+		case KEY_F9:
+		case KEY_F10:
+		case KEY_F11:
+		case KEY_F12:
+		case KEY_F13:
+		case KEY_F14:
+		case KEY_F15:
+		case KEY_F16:
+		case KEY_F17:
+		case KEY_F18:
+		case KEY_F19:
+		case KEY_F20:
+		case KEY_F21:
+		case KEY_F22:
+		case KEY_F23:
+		case KEY_F24:
+		case KEY_NUMLOCK:
+		case KEY_SCROLL:
+		case KEY_LCONTROL:
+		case KEY_RCONTROL:
+		return true;
+
+	default:
+		return false;
+	}
+}

```

Comment:
I noticed this is pretty lengthy. Would it work to put these in an std::array and do a linear search (std::array::find)?

Reply:
That is possible, but it would not make the code any shorter unless:
1. It would go way off the screen with a very long line, or
2. It would arbitrarily start newlines in some places.

Either way, it would break convention as all the other sections of code use switch cases. So I don't see that as an improvement.